assembl/views/api2/auth.py

```diff
@@ -546,7 +546,7 @@ def assembl_register_user(request):
                 print "email token:", request.route_url(
                     'user_confirm_email', token=email_token(account))
             if discussion:
-                check_subscription = discussion.preferences['whitelist_on_login']
+                check_subscription = discussion.preferences['whitelist_on_register'] or False
```

Comment:
default does not go here, but in prefs where I had original line note

Reply:
Ahh when you set a default, does the dict always get populated ?

Reply:
yes

Reply:
Fixed on fcef653

Reply:
more accurately: if not defined in the json, the getters cascade to the default value.